core/src/main/java/com/graphhopper/routing/ch/PrepareCHEdgeIteratorImpl.java

```diff
@@ -42,7 +60,48 @@ public PrepareCHEdgeIteratorImpl setBaseNode(int node) {
     @Override
     public boolean next() {
         assertBaseNodeSet();
-        return chIterator.next();
+        while (true) {
+            boolean hasNext = chIterator.next();
+            if (!hasNext) {
+                return false;
+            } else if (hasAccess()) {
+                return true;
+            }
+        }
+    }
+
+    private boolean hasAccess() {
+        if (isShortcut()) {
+            return shortcutFilter.accept((CHEdgeIterator) chIterator);
+        } else {
+            // c.f. comment in DefaultEdgeFilter
+            if (chIterator.getBaseNode() == chIterator.getAdjNode()) {
+                return finiteWeight(false) || finiteWeight(true);
+            }
+            return shortcutFilter.fwd && finiteWeight(false) || shortcutFilter.bwd && finiteWeight(true);
+        }
+    }
+
+    private boolean finiteWeight(boolean reverse) {
+        return !Double.isInfinite(getOrigEdgeWeight(reverse, false));
+    }
+
+    /**
+     * @param needWeight if true this method will return as soon as its clear that the weight is finite (no need to
+     *                   do the full computation)
+     */
+    private double getOrigEdgeWeight(boolean reverse, boolean needWeight) {
+        // todo: for #1776 move the access check into the weighting
+        final boolean access = reverse
+                ? chIterator.getReverse(accessEnc)
+                : chIterator.get(accessEnc);
+        if (!access) {
+            return Double.POSITIVE_INFINITY;
+        }
+        if (!needWeight) {
+            return 0;
+        }
+        return weighting.calcWeight(chIterator, reverse, EdgeIterator.NO_EDGE);
```

Comment:
I thought about this isFinite method too, but I do not think we should introduce this. Maybe as GHUtility but not for Weighting. 

If this is provided from the Weighting it is suggested we can separate soft from hard constraints, but soft constraints can always degenerate into hard constraints e.g. too big factors in multiplications leading to an overflow into infinity. And so the isFinite method has to be always like Double.isFinite(calcWeight)

Reply:
We could make this clear in javadocs: isFinite would allow the creator of the weighting to benefit performance-wise but it would be wrong to 'break the contract' by returning something that does not correspond to `Double.isInfinite(calcWeight)`. Since the performance difference (for the current FastestWeighting at least) is relatively small we can continue without the method though and it should not be a big issue to add it later when its needed.

Reply:
The problem is that such a method would be hard to implement. The soft constraints (outside of Weighting.isFinite but inside calcWeight) need a very sensitive range check and even a combination of soft constraints should never return infinity i.e. if isFinite==false, then calcWeight<infinity.

But if this contract is violated we'll get the problems back that we currently have due to the access and weighting separation (#367, #1234). 

Or we introduce a Weighting.couldBeInfinity method that can be better used as pre-check of calcWeight.

Reply:
> Or we introduce a Weighting.couldBeInfinity method that can be better used as pre-check of calcWeight and used only for performance reasons.

Yes I think this is what I meant: Be able to fully block an edge (for example we might have a boolean encoded value set to false for all highways/Autobahn for bike). I would rather call it `isBlocked` ?

Reply:
`isBlocked` implies the separation that I would avoid. IMO we should call it `couldBeBlocked` :) ... I see a bigger problem here as the "implementor" is required to keep `isBlocked` and `calcWeight` consistent. Where the past showed that this is hard to achieve even from ourselves.

And `couldBeBlocked` could just use the current access bit stuff and then `calcWeight` can still return infinity.

But well, as you said, we probably should not introduce this right now as it is really only a performance tweak. And other (more evil) techniques like caching should be tried too ;)